client/landing/stepper/declarative-flow/internals/index.tsx

```diff
@@ -130,7 +135,10 @@ export const FlowRenderer: React.FC< { flow: Flow } > = ( { flow } ) => {
 		const isReEnteringStep =
 			signupCompleteFlowName === flow.name && signupCompleteStepName === currentStepRoute;
 		if ( ! isReEnteringStep ) {
-			recordStepStart( flow.name, kebabCase( currentStepRoute ), { intent } );
+			recordStepStart( flow.name, kebabCase( currentStepRoute ), {
+				intent,
+				assembler_source: getAssemblerSource( design ),
```

Comment:
Could we add the property when we have the design or event the current step is assembler? I assume this value is not general enough for all steps. WDTY?

Reply:
Thanks for the idea. I think it doesn't hurt and can be useful to have it on all steps, not only for the step name `patternAssembler`. WDYT @taipeicoder?

Reply:
I think makes sense to at least add it when `design` is defined, but I'm fine with it being available in non-assembler steps. Your call 🙂
```suggestion
				...( design && { assembler_source: getAssemblerSource( design ) } ),
```

Reply:
Sounds good, added 👍